DENSE_utilities/DENSE_toolbox/AnalysisViewer.m

```diff
@@ -1538,44 +1538,59 @@ function deleteFcn()
 function [file, out] = exportMatFcn(obj,startpath)
 
     % check for startpath
-    if nargin < 2 || isempty(startpath) || ~exist(startpath,'dir')
+    if nargin < 2 || isempty(startpath)
         startpath = pwd;
     end
 
+    if ~exist(startpath, 'dir')
+        [origstartpath,~,ext] = fileparts(startpath);
+        file = startpath;
+        startpath = origstartpath;
+        if isempty(ext)
+            % If there is no extension then MATLAB save will add
+            % '.mat'. This will allow us to return the correct
+            % filename to the user.
+            file = [file '.mat'];
+        end
+    end
```

Comment:
This addresses the previous issue with extensions. MATLAB save has the nasty function of adding a '.mat' to the filename if no extension. We have two choices

1. Throw an error if there is no extension.
2. Set the extension manually so we return the right file name.

I went with 2 here, but could easily do 1, instead.